atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -110,13 +112,20 @@ protected int getPoolSize() {
                     log.debug("Returning {} to pool", resource);
 
                     // eagerly cleanup idle-connection read buffer to keep a smaller memory footprint
-                    TTransport transport = resource.getInputProtocol().getTransport();
-                    if (transport instanceof TMemoryInputTransport) {
-                        byte[] underlyingBuffer = transport.getBuffer();
-                        if (underlyingBuffer != null) {
-                            log.debug("During {} check-in, cleaned up an iprot of size {}", resource, underlyingBuffer.length);
-                            ((TMemoryInputTransport) transport).clear();
+                    try {
+                        TTransport transport = resource.getInputProtocol().getTransport();
+                        if (transport instanceof TFramedTransport) {
+                            Field readBuffer_ = ((TFramedTransport) transport).getClass().getDeclaredField("readBuffer_");
+                            readBuffer_.setAccessible(true);
+                            TMemoryInputTransport tMemoryInputTransport = (TMemoryInputTransport) readBuffer_.get(transport);
+                            byte[] underlyingBuffer = tMemoryInputTransport.getBuffer();
+                            if (underlyingBuffer != null) {
+                                log.debug("During {} check-in, cleaned up a read buffer of {} bytes", resource, underlyingBuffer.length);
+                                tMemoryInputTransport.clear();
```

Comment:
I realize upstream Thrift is unlikely to change, but would prefer putting `tMemoryInputTransport.clear();` outside the `if` to always clear.